core/search/boolean_filter.hpp

```diff
@@ -174,6 +180,7 @@ class IRESEARCH_API Or : public boolean_filter {
 
  private:
   size_t min_match_count_;
+  mutable size_t optimized_match_count_{0}; // count of always matched filters optimized away
```

Comment:
all filters are meant to be treated as builders, we're not allowed to store internal mutable state